app/src/main/java/org/luxoft/sdl_core/ClassicBtHandler.java

```diff
@@ -0,0 +1,375 @@
+                String deviceName = device.getName();
+                String deviceHardwareAddress = device.getAddress();
+                Log.i(TAG, "Classic BT device is found " + deviceName + " with address " + deviceHardwareAddress);
+
+                connect(device);
+            }
+        }
+    };
+
+
+    /**
+     * This thread runs while attempting to make an outgoing connection
+     * with a device. It runs straight through; the connection either
+     * succeeds or fails.
+     */
+    private class ConnectThread extends Thread {
+        private final BluetoothSocket mmSocket;
+        private final BluetoothDevice mmDevice;
+
+        public ConnectThread(BluetoothDevice device) {
+            mmDevice = device;
+            BluetoothSocket tmp = null;
+
+            // Get a BluetoothSocket for a connection with the
+            // given BluetoothDevice
+            try {
+                tmp = device.createInsecureRfcommSocketToServiceRecord(SDL_UUID);
+            } catch (IOException e) {
+                Log.e(TAG, "Socket create() failed", e);
+            }
+            mmSocket = tmp;
+            mState = STATE_CONNECTING;
+        }
+
+        public void run() {
+            Log.i(TAG, "BEGIN mConnectThread");
+
+            // Always cancel discovery because it will slow down a connection
+            mBtAdapter.cancelDiscovery();
+
+            // Make a connection to the BluetoothSocket
+            try {
+                // This is a blocking call and will only return on a
+                // successful connection or an exception
+                mmSocket.connect();
+            } catch (IOException e) {
+                // Close the socket
+                try {
+                    mmSocket.close();
+                } catch (IOException e2) {
+                    Log.e(TAG, "unable to close() socket during" +
+                            "connection failure", e2);
+                }
+                //connectionFailed();
+                return;
+            }
+
+            // Reset the ConnectThread because we're done
+            synchronized (ClassicBtHandler.this) {
+                mConnectThread = null;
+            }
+
+            Log.d(TAG, "Start the connected thread!!!");
+            // Start the connected thread
+            connected(mmSocket, mmDevice);
+        }
+
+        public void cancel() {
+            try {
+                mmSocket.close();
+            } catch (IOException e) {
+                Log.e(TAG, "close() of connect socket failed", e);
+            }
+        }
+    }
+
+    /**
+     * Start the ConnectedThread to begin managing a Bluetooth connection
+     *
+     * @param socket The BluetoothSocket on which the connection was made
+     * @param device The BluetoothDevice that has been connected
+     */
+    public synchronized void connected(BluetoothSocket socket, BluetoothDevice
```

Comment:
@ychernysheva bad name for a function. Please rename it to something more obvious
If this is an override method, please add corresponding macro